ouroboros-consensus/src/Ouroboros/Consensus/MiniProtocol/ChainSync/Client.hs

```diff
@@ -82,19 +82,18 @@ type Consensus (client :: Type -> Type -> (Type -> Type) -> Type -> Type) blk m
    client (Header blk) (Tip blk) m ChainSyncClientResult
 
 -- | Abstract over the ChainDB
-data ChainDbView m blk = ChainDbView
-  { getCurrentChain   :: STM m (AnchoredFragment (Header blk))
-  , getCurrentLedger  :: STM m (ExtLedgerState blk)
-  , getOurTip         :: STM m (Tip blk)
-  , getIsInvalidBlock :: STM m (WithFingerprint (HeaderHash blk -> Maybe (InvalidBlockReason blk)))
-  }
+data ChainDbView m blk = ChainDbView {
+      getCurrentChain   :: STM m (AnchoredFragment (Header blk))
+    , getCurrentLedger  :: STM m (ExtLedgerState blk)
```

Comment:
Do we still need `getCurrentLedger`?

Reply:
We could avoid it, but it's not worth it: we use it in `findIntersection`, in the same transaction as `getCurrentChain` to get the `HeaderState` corresponding to the current tip.